subsys/fs/littlefs_fs.c

```diff
@@ -60,8 +64,7 @@ static K_HEAP_DEFINE(file_cache_heap, CONFIG_FS_LITTLEFS_FC_HEAP_SIZE);
 
 static inline bool littlefs_on_blkdev(struct fs_mount_t *mountp)
 {
-	return IS_ENABLED(CONFIG_FS_LITTLEFS_BLK_DEV) &&
-		mountp->flags & FS_MOUNT_FLAG_USE_DISK_ACCESS;
+	return mountp->flags & FS_MOUNT_FLAG_USE_DISK_ACCESS;
```

Comment:
How tough is that flag specification?

If I may argue here - automatically switching from flash storage to disk access and vice versa is masking potential fault in user code and it is unwanted behavior on system where both are made available. If third interface is added in the future, the logic gets unnecessarily confusing, but it should be simple and straightforward.

I listed all combinations by the spec, by the old code and new code:
![image](https://user-images.githubusercontent.com/91126161/198218514-d2c0cebd-9f9c-4efb-b59d-c5b37505c34d.png)

6th combination (green cells) changes the behavior. It would be a good time to change the spec and behavior if wanted because later change causes also 3rd combination change.

Reply:
OK then, we should change how the flag works, this requires stable API change process and changes to FAT FS too.